functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+# Callback Functions
+
+## Goal
+
+One of the ways we'll solve problems in JavaScript is using callback functions. Getting the hang of callback functions takes practice.
+
+The goal of this lesson is to introduce callback functions, so we can use this vocabulary as we see functions passed around as arguments in the future.
+
+## Another Way to Look at Functions
+
+We might have been in situations before where we thought, "Ugh! This logic is reusable. The only thing that needs to change is one piece of behavior." If we've ever thought, "If only we could take in behavior as an argument," we're in luck!
+
+For example, let's imagine this problem and pseudocode:
+
+We need to build one calculator function that can add, subtract, multiply, and divide two numbers.
+
+In order to accomplish that goal, we need three arguments:
+
+1. The first number
+2. The second number
+3. Something that represents to operation we need, to help us determine if we're adding, subtracting, multiplying, or dividing
+
+We could approach the third argument by expecting a string, such as `'add'`, `'+'`, `'multiply'`, or `'*'`.
+
+<!-- prettier-ignore-start -->
+```js
+const calculator = function(a, b, operation){
+  // if (operation === 'add') {
+  //  return a + b;
+  // }
+}
+```
+<!-- prettier-ignore-end -->
+
+However, wouldn't it be grand if we could _pass in_ the actual operation, whether it's addition or subtraction, and execute it?
+
+## Callback Functions
+
+In order to pass around behavior, we pass around functions.
+
+More specifically, a function can _take in a function_ as an _argument_, and then call that function.
+
+To help us talk about functions, we can use specific vocabulary around these functions. A **callback function** is a nickname for a function that is _passed into_ another function.
+
+We can reverse that definition and imagine a function that _takes in another function as an argument_. The function passed in as an argument can be labeled as a _callback function_.
+
+### Passing Around Cleaning Up Behavior
+
+Let's dive into an example! Let's define these two functions, `cleanKitchen` and `bakeCake`.
+
+<!-- prettier-ignore-start -->
+```js
+const cleanCountertop = function () {
+  console.log('I\'m cleanin\' my counters now! Scrubba scrub scrub.');
+}
+
+const bakeCake = function () {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+}
+
+bakeCake();
+```
+<!-- prettier-ignore-end -->
+
+The variable `cleanCountertop` references a function that prints details about cleaning a kitchen.
+
+The variable `bakeCake` references a different function that prints details about baking a cake.
+
+When we bake a cake, maybe there's always cleanup behavior that we want to call, whether it's cleaning our counters, sink, or stove-top.
```

Comment:
Update this description to reflect the additional motivation I mentioned before: customizing an algorithm mid-step.

Of course there are plenty of places in JS use that we _will_ provide a continuation function that runs at the end (especially after async stuff) but since we can't really see that here, I think this is a little more compelling.

```suggestion
As part of baking a cake, we need to wait while the cake is in the oven. Let's assume we always want to perform some kind of cleanup behavior while we're waiting, whether it's cleaning our counters, sink, or stove-top.

We can't just call our `cleanCountertop` before we start baking the cake, or after we're done baking the cake. We want to do it in the middle, while the cake is in the oven!
```